src/main/java/uk/gov/hmcts/reform/prl/services/hearingmanagement/HearingManagementService.java

```diff
@@ -0,0 +1,99 @@
+package uk.gov.hmcts.reform.prl.services.hearingmanagement;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDataContent;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.ccd.client.model.Event;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+import uk.gov.hmcts.reform.prl.models.dto.ccd.CaseData;
+import uk.gov.hmcts.reform.prl.models.dto.hearingmanagement.HearingRequest;
+import uk.gov.hmcts.reform.prl.services.AuthorisationService;
+import uk.gov.hmcts.reform.prl.services.SystemUserService;
+import uk.gov.hmcts.reform.prl.utils.CaseUtils;
+
+import static uk.gov.hmcts.reform.prl.constants.PrlAppsConstants.CASE_TYPE;
+import static uk.gov.hmcts.reform.prl.constants.PrlAppsConstants.JURISDICTION;
+
+@Slf4j
+@RestController
+@SecurityRequirement(name = "Bearer Authentication")
+public class HearingManagementService {
+
+    private static final String HMC_STATUS = "LISTED";
+    public static final String HEARING_STATE_CHANGE_SUCCESS = "hmcCaseUpdateSuccess";
+    public static final String HEARING_STATE_CHANGE_FAILURE = "hmcCaseUpdateFailure";
+
+    @Autowired
+    private AuthorisationService authorisationService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private SystemUserService systemUserService;
+
+    @Autowired
+    private CoreCaseDataApi coreCaseDataApi;
+
+    public void stateChangeForHearingManagement(HearingRequest hearingRequest, String s2sToken) {
+
+        log.info("Processing the callback for the caseId {} with HMC status {}", hearingRequest.getCaseRef(),
+                     hearingRequest.getHearingUpdate().getHmcStatus());
+
+        String userToken = systemUserService.getSysUserToken();
+        String systemUpdateUserId = systemUserService.getUserId(userToken);
+        log.info("Fetching the Case details based on caseId {}", hearingRequest.getCaseRef()
+        );
+
+        CaseDetails caseDetails = coreCaseDataApi.getCase(
+            userToken,
+            s2sToken,
+            hearingRequest.getCaseRef()
+        );
+        CaseData caseData = CaseUtils.getCaseData(caseDetails, objectMapper);
+
```

Comment:
Change this to switch / if condition based on hearing status. if the hearing status is 
LISTED - Change the state from PREPARED_FOR_HEARING to DECISION_OUTCOME
ADJOURNED / POSTPONED - Don't change the state and keep the status as PREPARED_FOR_HEARING
CANCELLED - Keep a placeholder for this to be developed later